main.js

```diff
@@ -0,0 +1,159 @@
+                if(VALIDATE(arr[i]) || validateNumbers(arr[i])) {
+                    return;
+                }
+            }
+        } console.log(arr);
+}
+createArray(arraySize, newArray);
+
+// 2. Відсортувати масив за зростанням.
+newArray.sort((num_1, num_2) => {
+    return num_1 - num_2;
+});
+console.log(newArray);
+
+
+// 3. Дано масив - список покупок. Кожен елемент масиву - це обʼєкт вигляду: {productName: 'bread', productPrice: 14.5, productQuantity: 2}. Мінімальний довжина такого масиву - 6
+
+const productArray = [
+    {productName: "молоко", productPrice: 35, productQuantity: 2},
+    {productName: "сир", productPrice: 100, productQuantity: 5},
+    {productName: "хліб", productPrice: 35, productQuantity: 4},
+    {productName: "м'ясо", productPrice: 150, productQuantity: 3},
+    {productName: "виноград", productPrice: 85, productQuantity: 1},
+    {productName: "авокадо", productPrice: 60, productQuantity: 4},
+    {productName: "рисова локшина", productPrice: 200, productQuantity: 2},
+];
+// Created a function to sort the array by field.
+const sortedByField = (field) => {
+    return (a, b) => a[field] > b[field] ? 1 : -1;
+};
+
+// Порахувати загальну вартість корзини та вивести суму у зрозумілому форматі
+const totalPrice = productArray.reduce((totalPrice, {productPrice, productQuantity}) => totalPrice + productPrice * productQuantity, 0);
+console.log(`Загальна вартість продуктової корзини складає ${totalPrice} грн.`);
+
+// Порахувати найменшу кількість продукту, який потрібно купити
+const sortedByQuantity = productArray.sort(sortedByField("productQuantity"));
+console.log(sortedByQuantity)
+// console.log("Найменша кількість продукту, який ми маємо придбати - " + sortedByQuantity[0].productName + " - " + sortedByQuantity[0].productQuantity + ".")
+
+// Порахувати загальну кількість продуктів
+const totalQantity = productArray.reduce((totalQantity, {productQuantity}) => totalQantity + productQuantity, 0);
+console.log(`Загальна кількість товарів = ${totalQantity}.`)
+
+// Знайти найдорожчий продукт
+const mostExpensiveProduct = productArray.sort(sortedByField("productPrice"));
+console.log("Найдорожчий продукт, який ми маємо придбати - " + mostExpensiveProduct[mostExpensiveProduct.length-1].productName + ", продукт коштує - " + mostExpensiveProduct[mostExpensiveProduct.length-1].productPrice + " грн.")
+
+// Створити функцію, яка повинна додавати новий продукт в існуючий масив
+const newProduct = {};
+
+const addNewProduct = (array, obj) => {
+    obj.productName = prompt("Введіть назву продукту, який би додатково хотіли б придбати.");
+    obj.productPrice = +prompt("Вкажіть вартість цього продукту.");
+    obj.productQuantity = +prompt("Вкажіть кількість цього продукту.");
+    if (VALIDATE(obj.productName) || !isNaN(obj.productName) || VALIDATE(obj.productPrice) || validateNumbers(obj.productPrice) || VALIDATE(obj.productQuantity) || validateNumbers(obj.productQuantity)){
```

Comment:
`!isNaN(obj.productName)` why do we need to check is string is NaN?
Also, there is no validation for price

<img width="609" alt="image" src="https://user-images.githubusercontent.com/84010501/193127326-49c996df-34a3-49cc-8f79-3eb3ba5016f9.png">

Reply:
It still works with strings
<img width="540" alt="image" src="https://user-images.githubusercontent.com/84010501/194806762-41b674ea-1155-4390-baa7-57d24d5a3de6.png">